### ISSUE TITLE
prefer-const generates false positive (same scope)

### ISSUE THREAD

**Issue by mdebruijne** [eslint/eslint#5350]
There already is an issue report that prefer-const generates false positives when variables are declared in a higher scope (issue #5251), but also in the same scope prefer-const generates false positives.

Code (simplified);

```js
'use strict'

/* eslint prefer-const: 2 */
/* eslint-env es6 */

const myFunction = () => {
  let myVar

  myVar = 'myValue'
  console.log(myVar)
}

myFunction()
```

ESLint output;
```sh
7:7  error  'myVar' is never modified, use 'const' instead  prefer-const
```

ESLint config;
```
none, only inline config
```

Versions;
```sh
eslint --version
v2.2.0

node --version
v5.6.0
```

If you need more information then please let me know.

**Comment by michaelficarra** [comment]
What makes you think that you can't replace that with a const?

**Comment by ilyavolodin** [comment]
@mdebruijne You example can be safely re-written with
```js
'use strict'

/* eslint prefer-const: 2 */
/* eslint-env es6 */

const myFunction = () => {
  const myVar = 'myValue';

  console.log(myVar);
};

myFunction();
```

**Comment by mdebruijne** [comment]
@michaelficarra, @ilyavolodin 

I am aware that this can be rewritten to ```const myVar = 'myValue'```

This simplified example demonstrates the incorrect error message ```error  'myVar' is never modified```. The error message is incorrect (and therefore confusing) as the myVar variable is modified in the example provided. The error should probably be changed to something like ```error  'let myVar' can be refactored to 'const myVar'```.

If you agree to this change then I can send a pull request if desired.

**Comment by michaelficarra** [comment]
:-1: This rule is talking about reassignment. You've simply separated declaration from initialisation, but you've still not used any reassignment. I'd be in favour of changing the wording from "modified" to "reassigned".

**Comment by mdebruijne** [comment]
Agree. Changing "modified" to "reassigned" better reflects the reasoning behind this rule and makes the error message less confusing.

Pull request required?

**Comment by platinumazure** [comment]
@mdebruijne I would advise waiting for the issue to get an "accepted" label, but then yes, you may submit a pull request if you like. I highly doubt this is something the core team would jump on since there are higher priorities.

**Comment by ilyavolodin** [comment]
modified -> reassigned sounds good to me. PRs welcome.
